wagtail/admin/tests/test_templatetags.py

```diff
@@ -31,6 +33,67 @@
 from wagtail.utils.deprecation import RemovedInWagtail70Warning
 
 
+class TestAvatarUrlInterceptTemplateTag(WagtailTestUtils, TestCase):
+    def setUp(self):
+        self.get_avatar_hook_mock = mock.MagicMock()
+        self.test_user = self.create_user(
+            username="testuser",
+            email="testuser@email.com",
+            password="password",
+        )
+
+    def tearDown(self):
+        self.get_avatar_hook_mock.reset_mock()
+
+    def test_get_avatar_url_hook_undefined(self):
+        url = avatar_url(self.test_user)
+        self.assertFalse(self.get_avatar_hook_mock.called)
+        self.assertIn("www.gravatar.com", url)
+
+    @mock.patch("wagtail.admin.templatetags.wagtailadmin_tags.hooks")
+    def test_get_avatar_url_empty_value(self, mock_hooks):
+        self.get_avatar_hook_mock.return_value = None
+        mock_hooks.get_hooks.return_value = [self.get_avatar_hook_mock]
+
+        with hooks.register_temporarily("get_avatar_url", self.get_avatar_hook_mock):
```

Comment:
I'm not sure why you need to make so much use of mocks here - it makes it hard to figure out what piece of _real_ code is being tested here. In `testapp/wagtail_hooks.py` you've defined a real hook function that returns `/some/avatar/fred.png` (under some conditions), to be returned by the real `get_hooks` function - but here you're setting up a fake version of `get_hooks` to return a fake version of the hook function that behaves just like the real one. Also, if you've swapped out `get_hooks` then this call to `hooks.register_temporarily` is redundant, since it's modifying the behaviour of the real one.

I think all of this can be replaced with just two tests, with no mocks required - one where the hook function intercepts the default avatar logic, and one where it doesn't. (And that hook function can be defined either globally in testapp/wagtail_hooks.py or locally via `hooks.register_temporarily` - it doesn't need to be both.)